Source/Public/Get-VSTeamProcess.ps1

```diff
@@ -29,28 +30,25 @@ function Get-VSTeamProcess {
 
          Write-Output $project
       }
-      elseif ($Name) {
-         # Lookup Process ID by Name
-         Get-VSTeamProcess | where-object { $_.name -eq $Name }
-      }
+
       else {
          # Return list of processes
          try {
             # Call the REST API
-            $resp = _callAPI -area 'process' -resource 'processes' `
-               -Version $(_getApiVersion Core) -NoProject `
+            $resp = _callAPI -Area 'work' -resource 'processes' `
+               -Version $(_getApiVersion Graph)  -NoProject `
                -QueryString @{
                '$top'  = $top
                '$skip' = $skip
             }
 
-            $objs = @()
-            
-            foreach ($item in $resp.value) {
-               $objs += [VSTeamProcess]::new($item)
-            }
-
-            Write-Output $objs
+           #we just fetched the processes so let's update the cache. Also Cache the URLS for processes
+            [VSTeamProcessCache]::processes = $resp.value.name | Sort-Object

```

Comment:
Something about updating the cache using the $Name filter seems like a bug. If the Name filter, filters the results to a single record that will be the only record in the cache. That would be a problem. All calls until the cache expires would only return that one record in the cache. The cache should not apply a Name filter.

Reply:
Would be nice if [VSTeamProcessCache]::Update was written to take an optional $list parameter. That way the $resp.value could be passed into [VSTeamProcessCache]::Update in this case. If Update is called with no parameters Get-VSTeamProcess would get called and update the cache with the full list of processes.

Looks like Update is expecting VSTeamProcess objects instead of the raw object from the service... So I guess you cloud

```
foreach ($item in $resp.value) {
   $objs += [VSTeamProcess]::new($item)
}

[VSTeamProcessCache]::Update -List $objs

return $objs |  Where-Object { $_.name -like $Name } | Sort-Object -Property Name
```

Reply:
> Would be nice if [VSTeamProcessCache]::Update was written to take an optional $list parameter. That way the $resp.value could be passed into [VSTeamProcessCache]::Update in this case. 

Yes, that's the direction I'm going in. If these classes switch to C# then the content and timestamp would become hidden and you call "Invalidate" ,"IsExpired", "Update()" and/or "Update(content)".  I'm also moving towards the cache using Get-Process (not _getProcesses) but since that does a validate, which checks the cache it must ensure the validate returns when the parameter is empty without looking at the cache first otherwise it gets into an infinite loop (as I discovered!)